### PR TITLE
libzeth: fix segfault in filesystem_util

### PR DESCRIPTION
**Pull request by dtebbs**
fix segfault under some circumstances
rearranged code slightly so potentially null var is only handled in confined set of lines.

### CODE REVIEW

File: libzeth/serialization/filesystem_util.cpp [hunk 1]
```diff
@@ -10,33 +10,33 @@ namespace libzeth
...
-    if (nullptr == path) {
-        // Fallback destination if the ZETH_TRUSTED_SETUP_DIR env var is not set
-        // We assume below that `std::getenv("HOME")` does not return `nullptr`
-        boost::filesystem::path home_path =
-            boost::filesystem::path(std::getenv("HOME"));
-        boost::filesystem::path zeth_setup("zeth_setup");
-        boost::filesystem::path default_path = home_path / zeth_setup;
-
-        return default_path;
+    if (nullptr != path) {
```
Comment (AntoineRondelet) [inline libzeth/serialization/filesystem_util.cpp:13]: Let's invert the 2 `path != nullptr`

**Comment by AntoineRondelet** [comment]
Good catch @dtebbs, a return was indeed missing in one of the block. Thanks!

File: libzeth/serialization/filesystem_util.cpp (hunk 1 above)
Comment (dtebbs) [reply libzeth/serialization/filesystem_util.cpp]: updated

Comment (AntoineRondelet) [reply libzeth/serialization/filesystem_util.cpp]: Thanks

**Pull request merged**